packages/utxo-lib/src/bip32.ts

```diff
@@ -131,13 +132,15 @@ class BIP32 implements BIP32Interface {
     }
 
     get publicKey(): Buffer {
-        if (this.__Q === undefined) this.__Q = ecc.pointFromScalar(this.__D, true);
+        if (this.__Q === undefined && this.__D !== undefined) {
+            this.__Q = uin8ArrayToBuffer(ecc.pointFromScalar(this.__D, true)) ?? undefined;
+        }
 
         return this.__Q!;
     }
 
     get privateKey(): Buffer | undefined {
-        return this.__D;
+        return uin8ArrayToBuffer(this.__D) ?? undefined;
```

Comment:
is this needed? `this.__D` is defined as Buffer and should be a Buffer.

this comment applies for all the usage of uin8ArrayToBuffer in this file